### PR TITLE
XYChart: Remove old implementation

### PR DESCRIPTION
**Pull request by leeoniya**
supersedes https://github.com/grafana/grafana/pull/90802

this moves everything from the `v2` subdir into the top-level `xychart` dir and replaces everything that was there.

we've been rolled out with new implementation [for several months](https://ops.grafana-ops.net/d/5mFErDb7z/grafana-features-rollout-statuses?orgId=1&var-Feature=autoMigrateXYChartPanel&var-Version=1%5B01%5D%5C..%2A&from=now-90d&to=now&timezone=Europe%2FBerlin); it's time.

this gdev dashboard tests that migrations from original schema continue to work:
http://localhost:3000/d/YObbMr44k/panel-tests-xy-chart-migrations?orgId=1

this gdev dashboard tests the new schema:
http://localhost:3000/d/fdn48fmz8f94wc/panel-tests-xy-chart-demo?orgId=1

### CODE REVIEW

File: public/app/plugins/panel/xychart/panelcfgold.gen.ts
Comment (leeoniya) [inline public/app/plugins/panel/xychart/panelcfgold.gen.ts:1]: keeping this around to keep the migration code understandable and `any`-free

File: public/app/plugins/panel/xychart/plugin.json [hunk 1]
```diff
@@ -2,7 +2,6 @@
   "type": "panel",
   "name": "XY Chart",
   "id": "xychart",
-  "state": "beta",
```
Comment (leeoniya) [inline public/app/plugins/panel/xychart/plugin.json:5]: GA :tada:

File: public/app/features/plugins/built_in_plugins.ts [hunk 2]
```diff
@@ -67,13 +65,7 @@ const welcomeBanner = async () =>
...
 const canvasPanel = async () => await import(/* webpackChunkName: "canvasPanel" */ 'app/plugins/panel/canvas/module');
 const graphPanel = async () => await import(/* webpackChunkName: "graphPlugin" */ 'app/plugins/panel/graph/module');
-const xychartPanel = async () => {
-  if (config.featureToggles.autoMigrateXYChartPanel) {
-    return await import(/* webpackChunkName: "xychart2" */ 'app/plugins/panel/xychart/v2/module');
-  } else {
-    return await import(/* webpackChunkName: "xychart" */ 'app/plugins/panel/xychart/module');
-  }
-};
+const xychartPanel = async () => await import(/* webpackChunkName: "xychart" */ 'app/plugins/panel/xychart/module');
```
Comment (leeoniya) [inline public/app/features/plugins/built_in_plugins.ts:68]: image

File: pkg/services/featuremgmt/registry.go [hunk 3]
```diff
@@ -142,14 +142,6 @@ var (
...
 			Owner:        grafanaDatavizSquad,
 		},
-		{
-			Name:         "autoMigrateXYChartPanel",
-			Description:  "Migrate old XYChart panel to new XYChart2 model",
-			Stage:        FeatureStageGeneralAvailability,
-			FrontendOnly: true,
-			Expression:   "true", // enabled by default
-			Owner:        grafanaDatavizSquad,
-		},
```
Comment (leeoniya) [inline pkg/services/featuremgmt/registry.go:152]: :wave: cya!

**Review by gelicia — APPROVED** [review]
Found a couple things

- Err as the message when you first enter an XY with invalid data isn't consistent with what we see in other visualizations
- Toggling legend visibility or changing any legend settings causes the data to disappear until you save and do a window refresh

I'm assuming since this has been on by default for a while, these are not new. I'll write up an issue ticket for them.

I looked through the escalations and found one bug we fixed, and one case where we advised the customer to force-disable the new chart. This customer will be unable to do so going forward. We ultimately advised the customer to use the trend panel instead, and the issue was closed. This should be a good solution going forward if they force-disabled the feature flag.

File: public/app/plugins/panel/xychart/module.tsx [hunk 4]
```diff
@@ -1,41 +1,35 @@
...
+import { SeriesEditor } from './SeriesEditor';
+import { XYChartPanel2 } from './XYChartPanel';
 import { getScatterFieldConfig } from './config';
-import { Options, FieldConfig, defaultFieldConfig } from './panelcfg.gen';
+import { xyChartMigrationHandler } from './migrations';
+import { FieldConfig, defaultFieldConfig, Options } from './panelcfg.gen';
 
-export const plugin = new PanelPlugin<Options, FieldConfig>(XYChartPanel)
+export const plugin = new PanelPlugin<Options, FieldConfig>(XYChartPanel2)
+  // .setPanelChangeHandler(xyChartChangeHandler)
```
Comment (nmarrs) [inline public/app/plugins/panel/xychart/module.tsx:11]: Probably can delete this line too

File: public/app/plugins/panel/xychart/utils.ts [hunk 5]
```diff
@@ -7,3 +26,301 @@ export function fmt(field: Field, val: number): string {
...
+    );
+    let yMatcher = getFieldMatcher(
+      seriesCfg.y?.matcher ?? {
+        id: FieldMatcherID.byType,
+        options: 'number',
+      }
+    );
+    let colorMatcher = seriesCfg.color ? getFieldMatcher(seriesCfg.color.matcher) : null;
+    let sizeMatcher = seriesCfg.size ? getFieldMatcher(seriesCfg.size.matcher) : null;
+    // let frameMatcher = seriesCfg.frame ? getFrameMatchers(seriesCfg.frame) : null;
```
Comment (nmarrs) [inline public/app/plugins/panel/xychart/utils.ts:77]: another spot we can probs remove the old commented out implementation :)

File: public/app/plugins/panel/xychart/utils.ts [hunk 6]
```diff
@@ -7,3 +26,301 @@ export function fmt(field: Field, val: number): string {
...
+      if (s.size.field == null) {
+        // derive fixed size from y field config
+        s.size.fixed = s.y.field.config.custom.pointSize?.fixed ?? 5;
+        // ser.size.mode =
+      }
+    });
+
+    autoNameSeries(series);
+
+    // TODO: re-assign y display names?
```
Comment (nmarrs) [inline public/app/plugins/panel/xychart/utils.ts:226]: Is this still needed?

**Review by nmarrs — COMMENTED** [review]
tested gdev dashboards locally and they are looking good overall :)

a few minor nits on considering cleaning up and removing commented code blocks

I also noticed a minor UX issue where we don't seem to filter out the X + Y fields for the size + color fields when they automatically selected, we do for the X field once it has been selected but strangely not for the Y when it is selected which can lead to breaking the panel - in general a review of how the options interact with each other and providing some useful err messages when config is incorrect might be a good idea :) (ideal state though being that users can't break it in the first place :s) 

https://github.com/user-attachments/assets/d664eaa4-3723-415a-b2cc-d83432c3cf0c

**Pull request merged**